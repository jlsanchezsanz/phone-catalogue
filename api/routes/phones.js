const phonesRoutes = (app, fs) => {
  const dataPath = './data/phones.json';

  app.get('/phones', (_, res) => {
    fs.readFile(dataPath, 'utf8', (err, data) => {
      setTimeout(() => {
        if (err) {
          throw err;
        }
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.send(JSON.parse(data));
      }, 3000);
    });
  });
};

module.exports = phonesRoutes;
