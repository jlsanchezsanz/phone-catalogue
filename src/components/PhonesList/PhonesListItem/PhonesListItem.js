import React from 'react';

export default function PhonesListItem(props) {
  return (
    <div>
      <img
        className='image'
        src={require(`../../../assets/${props.phone.imageFileName}`)}
        alt={props.phone.name}
      />
      <h1 className='display-4'>
        {props.phone.manufacturer} {props.phone.name}
      </h1>
      <ul className='traits'>
        <li className='traits__item'>
          <span>{props.phone.price}â‚¬</span>
        </li>
        <li className='traits__item'>
          <span>{props.phone.ram}GB RAM</span>
        </li>
      </ul>
    </div>
  );
}
