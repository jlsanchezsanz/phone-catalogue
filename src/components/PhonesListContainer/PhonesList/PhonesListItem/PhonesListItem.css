.card {
  margin: 15px;
  width: 300px;
}

.card-body {
  height: 100px;
}

.phone__price {
  color: #326190;
  font-size: 30px;
  font-weight: 700;
}
