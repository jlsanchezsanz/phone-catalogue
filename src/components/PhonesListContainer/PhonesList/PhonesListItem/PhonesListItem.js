import React from 'react';
import PropTypes from 'prop-types';

import { PhonePropType } from '../../../../prop-types/phone.prop-type';

export default function PhonesListItem(props) {
  return (
    <div>
      <img
        className='image'
        src={require(`../../../../assets/${props.phone.imageFileName}`)}
        alt={props.phone.name}
      />
      <h1 className='display-4'>
        {props.phone.manufacturer} {props.phone.name}
      </h1>
      <ul className='traits'>
        <li className='traits__item'>
          <span>{props.phone.price}â‚¬</span>
        </li>
        <li className='traits__item'>
          <span>{props.phone.ram}GB RAM</span>
        </li>
      </ul>
    </div>
  );
}

PhonesListItem.propTypes = {
  phone: PropTypes.shape(PhonePropType).isRequired
};
