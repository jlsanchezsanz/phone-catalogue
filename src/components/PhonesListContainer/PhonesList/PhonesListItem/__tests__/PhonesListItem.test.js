import React from 'react';
import { shallow } from 'enzyme';

import PhonesListItem from '../PhonesListItem';
import { phonesMock } from '../../../../../mocks/phones.mock';

const setUpShallow = (phone) => {
  const component = shallow(<PhonesListItem phone={phone} />);
  return component;
};

describe('PhonesListItem', () => {
  let component;

  it('should display phones list item', () => {
    component = setUpShallow(phonesMock.data[0]);
    expect(component).toMatchSnapshot();
  });
});
