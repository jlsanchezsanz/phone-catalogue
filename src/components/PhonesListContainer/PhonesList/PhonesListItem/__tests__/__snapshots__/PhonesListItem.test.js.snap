// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhonesListItem should display phones list item 1`] = `
<div>
  <img
    alt="iPhone 7"
    className="image"
    src="IPhone_7.png"
  />
  <h1
    className="display-4"
  >
    Apple
     
    iPhone 7
  </h1>
  <ul
    className="traits"
  >
    <li
      className="traits__item"
    >
      <span>
        769
        â‚¬
      </span>
    </li>
    <li
      className="traits__item"
    >
      <span>
        2
        GB RAM
      </span>
    </li>
  </ul>
</div>
`;
